### PR TITLE
Printing speedup (from an hour to half an hour)

### PR DESCRIPTION
**Pull request by Retropotenza**
* Removed the workaround for the initial vertical position (fixed in the 1.2 release of Splatoon 2)
* Added bidirectional printing (the 1.2 release of Splatoon 2 fixed also the forth to last column skip, when moving from the right to the left)
* Removed time delays on echoes (verified that there is no need for them)
* Minimized the number of echoes (verified that at least 3 needed)
* Verified and cleaned up the USB control requests and the HID task (only the latter is needed, in it now return codes for endpoints reading and writing are checked)
* Pixel skipping happens only if the Switch detect a change on its HDMI connection (this happens even if the Switch accept incoming USB reports... I don't know if is possible to avoid this issue)

### CODE REVIEW

**Comment by undergroundmonorail** [comment]
This stops drawing when `ypos > 120`, but the bottom row of the image is y=119. As a result, the bottom row is overwritten with what I believe is just junk data that happens to be in ram.

I'm not super familiar with git so I don't know if I can make a pull request on your pull request or what, but changing line 224 from `if (ypos < 120)` to `if (ypos < 120 - 1)` should fix it (though I haven't tested it yet).

**Comment by Retropotenza** [comment]
Good catch, thanks 😁. In addition to this, the code will need to ink also when in `done` state, to print the dot at `x=0, y=319`.

I'll add these fixes in this pull request (adding a slightly modified test pattern), as soon as my final test will finish.

**Comment by LightningStalker** [comment]
Bidi, excellent!

File: README.md [hunk 1]
```diff
@@ -12,9 +12,9 @@ Unlike the Wii U, which handles these controllers on a 'per-game' basis, the Swi
 For my own personal use, I repurposed Switch-Fightstick to output a set sequence of inputs to systematically print Splatoon posts. This works by using the smallest size pen and D-pad inputs to plot out each pixel one-by-one.
 
 #### Printing Procedure
-Just press L to select the pixel pen and plug in the controller: it will automatically sync with the console, reset the cursor position and print. In case you see issues with controller conflicts while in docked mode, try using a USB-C to USB-A adapter in handheld mode. Printing currently takes about an hour.
+Just press L to select the pixel pen and plug in the controller: it will automatically sync with the console, reset the cursor position, clean the canvas and print. In case you see issues with controller conflicts while in docked mode, try using a USB-C to USB-A adapter in handheld mode. In dock mode, changes in the HDMI connection will briefly make the Switch not responding to incoming USB commands, skipping pixels in the printout. These changes may include turning the TV off, or switching the HDMI input in it (switching to the internal tuner will be OK, if this doesn't trigger a chanhe in the HDMI input).
```
Comment (LightningStalker) [inline README.md]: stop responding?

Comment (undergroundmonorail) [reply README.md]: s/chanhe/change/

File: Joystick.c [hunk 2]
```diff
@@ -197,88 +166,86 @@ void GetNextReport(USB_JoystickReport_Input_t* const ReportData) {
...
+			break;
+		case STOP_Y:
+			if (ypos < 120 - 1)
+				state = MOVE_Y;	
 			else
 				state = DONE;
-			break;
-		case MOVE_DOT:
-			if (xpos < 320 - 1)
+			break;
```
Comment (LightningStalker) [inline Joystick.c]: dat whitespace doe

**Comment by shinyquagsire23** [comment]
Oh nice, just let me know when you're ready for code review though since it seems there's still tweaking going on.

**Comment by Retropotenza** [comment]
It's should be ready... there was a bug when printing the last line, fixed (other commits was minor or typos).

**Comment by LightningStalker** [comment]
Yea, it's ready for prime time.  Just printed a post yesterday morning, it's working perfectly.

The only thing I still want to add is a pause function, but that's for another day.

**Comment by LightningStalker** [comment]
Just a couple tweaks to README.md

**Comment by shinyquagsire23** [comment]
LGTM, thanks!

**Pull request merged**